Assets/SEE/Controls/KeyActionDescriptor.cs

```diff
@@ -0,0 +1,57 @@
+﻿using UnityEngine;
+
+namespace SEE.Controls
+{
+    internal static partial class KeyBindings
+    {
+        /// <summary>
+        /// A descriptor of a user action that can be triggered by a key on the keyboard.
+        /// </summary>
+        internal class KeyActionDescriptor
+        {
+            /// <summary>
+            /// Constructor.
+            /// </summary>
+            /// <param name="name">descriptive name of the action shown to the user</param>
+            /// <param name="description">longer description of the action shown to the user</param>
+            /// <param name="category">category of the action</param>
+            /// <param name="keyCode">key this action is bound to</param>
+            public KeyActionDescriptor(string name, string description, KeyActionCategory category, KeyCode keyCode)
+            {
+                KeyCode = keyCode;
+                Name = name;
+                Category = category;
+                Description = description;
+            }
+
+            /// <summary>
+            /// The descriptive name of the action. It will be shown to the user. It should be
+            /// short and descriptive.
+            /// </summary>
+            public readonly string Name;
+            /// <summary>
+            /// A longer description of the action shown to the user. It provides more detail
+            /// than <see cref="Name"/>.
+            /// </summary>
+            public readonly string Description;
+            /// <summary>
+            /// <summary>
+            /// The scope of the action.
+            /// </summary>
+            public readonly KeyActionCategory Category;
+            /// The key on the keyboard this action is bound to. If that key is pressed,
+            /// the user wants to trigger this action.
+            /// </summary>
+            public KeyCode KeyCode;
+
+            /// <summary>
+            /// Returns this <see cref="KeyActionDescriptor"/> in a human-readable form.
+            /// </summary>
+            /// <returns>this <see cref="KeyActionDescriptor"/> as a meaningful string</returns>
+            public override string ToString()
+            {
+                return $"<KeyCode={KeyCode}, Name='{Name}', Category={Category}, Description='{Description}'>";
+            }
+        }
```

Comment:
Something that I feel would be very useful here to reduce the amount of boilerplate code is a recent C# feature: [**records**](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record). This whole class can be reduced to the following, with equal functionality:

```suggestion
        /// <param name="Name">The descriptive name of the action. It will be shown to the user. It should be
        /// short and descriptive.</param>
        /// <param name="Description">A longer description of the action shown to the user. It provides more detail
        /// than <see cref="Name"/>.</param>
        /// <param name="Category">The scope of the action.</param>
        /// <param name="KeyCode">The key on the keyboard this action is bound to. If that key is pressed,
        /// the user wants to trigger this action.</param>
        internal record KeyActionDescriptor(string Name, string Description, KeyActionCategory category, KeyCode keyCode) 
        {
            /// <summary>
            /// The key on the keyboard this action is bound to. If that key is pressed,
            /// the user wants to trigger this action.
            /// </summary>
            public KeyCode KeyCode { get; set; }
        }
```

With the following semantics:
* One readonly property per record parameter, with the documentation for each field given by the parameter's documentation. We declare `KeyCode` again to make its setter accessible.
* A constructor matching the parameters.
* Human-readable `ToString()` method, similar to the current one.
* Value equality, if two instances are ever compared.

Reply:
We need to be able to modify the `keyCode` field. In case of a `struct`, all its fields are read-only.

Reply:
An error I made in my original suggestion lead to a misunderstanding – the following should work as intended, including reference semantics (`record` as shorthand for `record class`) and a modifiable `KeyCode` field.
```suggestion
        /// <param name="Name">The descriptive name of the action. It will be shown to the user. It should be
        /// short and descriptive.</param>
        /// <param name="Description">A longer description of the action shown to the user. It provides more detail
        /// than <see cref="Name"/>.</param>
        /// <param name="Category">The scope of the action.</param>
        /// <param name="KeyCode">The key on the keyboard this action is bound to. If that key is pressed,
        /// the user wants to trigger this action.</param>
        internal record KeyActionDescriptor(string Name, string Description, KeyActionCategory Category, KeyCode KeyCode) 
        {
            /// <summary>
            /// The key on the keyboard this action is bound to. If that key is pressed,
            /// the user wants to trigger this action.
            /// </summary>
            public KeyCode KeyCode { get; set; }
        }
```